crates/ra_ide/src/completion/complete_keyword.rs

```diff
@@ -52,7 +53,28 @@ fn keyword(ctx: &CompletionContext, kw: &str, snippet: &str) -> CompletionItem {
     .build()
 }
 
+fn add_top_level_keywords(acc: &mut Completions, ctx: &CompletionContext) {
+    if let Some(token) = previous_non_triva_element(&ctx.token).and_then(|it| it.into_token()) {
+        if token.kind() == UNSAFE_KW {
+            acc.add(keyword(ctx, "impl", "impl $0 {}"));
+            acc.add(keyword(ctx, "trait", "trait $0 {}"));
+            acc.add(keyword(ctx, "fn", "fn $0() {}"));
+            return;
+        }
+    }
+    acc.add(keyword(ctx, "impl", "impl $0 {}"));
+    acc.add(keyword(ctx, "enum", "enum $0 {}"));
+    acc.add(keyword(ctx, "struct", "struct $0 {}"));
```

Comment:
```suggestion
    acc.add(keyword(ctx, "struct", "struct $0"));
```

As this  might be a unit struct. All the rest seem fine though.